analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantNullCheckCodeFixProvider.cs

```diff
@@ -24,62 +24,93 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using SonarAnalyzer.Extensions;
 using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [ExportCodeFixProvider(LanguageNames.CSharp)]
     public sealed class RedundantNullCheckCodeFixProvider : SonarCodeFixProvider
     {
         internal const string Title = "Remove this unnecessary null check";
-        public override ImmutableArray<string> FixableDiagnosticIds
-        {
-            get
-            {
-                return ImmutableArray.Create(RedundantNullCheck.DiagnosticId);
-            }
-        }
 
-        public override FixAllProvider GetFixAllProvider()
-        {
-            return WellKnownFixAllProviders.BatchFixer;
-        }
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(RedundantNullCheck.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
 
         protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
         {
             var diagnostic = context.Diagnostics.First();
             var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var diagnosticNode = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true);
 
-            if (!(root.FindNode(diagnosticSpan, getInnermostNodeForTie: true) is BinaryExpressionSyntax nullCheckNode))
+            if (diagnosticNode is BinaryExpressionSyntax nullCheckNode)
+            {
+                RegisterBinaryExpressionCodeFix(context, root, nullCheckNode);
+            }
+            else if (diagnosticNode is PrefixUnaryExpressionSyntax prefixUnary && prefixUnary.IsKind(SyntaxKind.LogicalNotExpression))
             {
-                return TaskHelper.CompletedTask;
+                RegisterBinaryExpressionCodeFix(context, root, prefixUnary);
+            }
+            else if (IsPatternExpressionSyntaxWrapper.IsInstance(diagnosticNode))
+            {
+                var isPatternExpression = (IsPatternExpressionSyntaxWrapper)diagnosticNode.RemoveParentheses();
+                if (isPatternExpression.IsNull() || isPatternExpression.IsNotNull())
```

Comment:
I think this is redundant, because anyway at this point we know a diagnostic has been raised on `diagnosticNode`, so it can only be `is null` or `is not null`

Reply:
Do you want to remove it or leave as defensive?

Reply:
I removed it.